File: output/ChatGPT/avatar/Python/C++/atcoder_ABC089_C.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    int N;
    cin >> N;
    vector<string> S(N);
    for (int i = 0; i < N; i++) {
        cin >> S[i];
    }
    
    vector<char> march = {'M', 'A', 'R', 'C', 'H'};
    vector<int> march_lis(5, 0);
    for (string s : S) {
        char c = toupper(s[0]);
        if (find(march.begin(), march.end(), c) != march.end()) {
            march_lis[find(march.begin(), march.end(), c) - march.begin()]++;
        }
    }
    
    long long ans = 0;
    for (int i = 0; i < 3; i++) {
        for (int j = i+1; j < 4; j++) {
            for (int k = j+1; k < 5; k++) {
                ans += march_lis[i] * march_lis[j] * march_lis[k];
            }
        }
    }
    
    cout << ans << endl;
    
}

File: output/ChatGPT/avatar/Python/Java/atcoder_ABC089_C.java
import java.util.Scanner;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;

public class atcoder_ABC089_C {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        int N = input.nextInt();
        String[] S = new String[N];
        for (int i = 0; i < N; i++) {
            S[i] = input.next();
        }
        ArrayList<String> march = new ArrayList<>(Arrays.asList("M", "A", "R", "C", "H"));
        int[] march_lis = new int[5];
        for (String s : S) {
            if (march.contains(Character.toString(s.charAt(0)).toUpperCase())) {
                march_lis[march.indexOf(Character.toString(s.charAt(0)).toUpperCase())]++;
            }
        }
        int ans = 0;
        for (int x : march_lis) {
            for (int y : march_lis) {
                for (int z : march_lis) {
                    if (x != y && y != z && x != z) {
                        ans += x * y * z;
                    }
                }
            }
        }
        System.out.println(ans);
    }
}